modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -240,13 +309,20 @@ public AssetLibrary patchAssetLibrary(
 		Map<String, String> descriptionMap = _getValue(
 			() -> LocalizedMapUtil.getI18nMap(group.getDescriptionMap()),
 			assetLibrary::getDescription_i18n);
+
+		if (assetLibrary.getSettings() == null) {
+			assetLibrary.setSettings(() -> new Settings());
+		}
+
 		UnicodeProperties unicodeProperties = _patchUnicodeProperties(
 			assetLibrary.getSettings(),
 			_getUnicodeProperties(
 				contextCompany.getCompanyId(),
 				group.getExternalReferenceCode()));
 
-		return _toAssetLibrary(
+		Permission[] permissions = assetLibrary.getPermissions();
```

Comment:
extracting here (like the autogenerated code) because the next `_toAssetLibrary` overwrite the "body" of the request

Reply:
Instead of reusing the variable, can you create a new one (e.g. `updatedAssetLibrary`)? This will make reordering things a bit safer.